Результат выполнения задания №2 и №3

SirDotJ opened the pull request:
Реализовал что видел в Figma формах. Попытался изменить немного svg волн чтобы они плавно переходили друг в друга при повторении по X. 
Второй аккаунт - работа над проектом с моего второго ноутбука.
В задании №3 попробоавал анимацию, но с Experience не успел, сообщите если необходимо доделать.

SirDotJ committed:
Загрузка используемых изображений

SirDotJ committed:
Типография полностью перенесена в соответствии с адаптивностью, небольшой рефакторинг

SirDotJ committed:
Начало оформление формы. Рабочий вариант header

Sir-DotJ committed:
Basic introduction section fillings + spelling apricot right

Sir-DotJ committed:
Basic experience section structure + Javascript spelling mistake fix

Sir-DotJ committed:
Basic experience section + apricot fix again.
Class structure to be thought more later.

Sir-DotJ committed:
Basic experience section + apricot fix again.
Class structure to be thought more later.

Sir-DotJ committed:
Basic work section

Sir-DotJ committed:
Testimonials in progress

SirDotJ committed:
Basic Testimonial structure and content

SirDotJ committed:
Basic Contact structure and content

SirDotJ committed:
Work on Introduction section: waves and appricotes till need work + refactor of fonts

SirDotJ committed:
Work on Skills: Typescript if off center, while carousel needs to be in the center + additional text modifications

SirDotJ committed:
Work on introduction waves, a bit on skills, still not on the center, start on experience

SirDotJ committed:
Work on Experience. No problems seen, ready for adaptibility.

SirDotJ committed:
Work on Work, skills are still missaligned, otherwise don't see issues

SirDotJ committed:
Work on testimonials. Ready for adaptibility.

SirDotJ committed:
Work on Contact. Ready for adaptibility.

SirDotJ committed:
Fixup of images in Work

SirDotJ committed:
Fixed the Skills alignment issue in a crutch way

SirDotJ committed:
Edited wave to make it repeatable, refactor, start of screen size based padding, header section is done, fixed work issues related to images, added viewport for correct @media usage.

SirDotJ committed:
Working waves

SirDotJ committed:
Introduction renamed to hero

SirDotJ committed:
Hero text gone on non desktop

SirDotJ committed:
Hero section is complete.

SirDotJ committed:
Experience is complete, minor fixes on hero

SirDotJ committed:
Experience is complete, minor fixes on hero

SirDotJ committed:
Work is complete. Not sure about image sizes but maybe they're fine.

SirDotJ committed:
Testimonials is done. Project is ready for deployment and subsequent cleanup.

SirDotJ committed:
Wave is visible, for review.

SirDotJ committed:
Fixed work's minimum width for description text issue.

SirDotJ committed:
Phone now gets copied on click

SirDotJ committed:
Email now gets copied to clipboard on click.

SirDotJ committed:
Scripts are put into separate js file under async.

SirDotJ committed:
Added pointer on elements that can be copied, fixed formatting issue for copied text.

SirDotJ committed:
Fixed missed gap difference on tablet-portrait and mobile screen sizes for Contact

SirDotJ committed:
Working animation for copying phone and email

SirDotJ committed:
Working burger button on mobile, awaiting animation.

SirDotJ committed:
Fixed issue from including burger div when selecting areas for padding in site

SirDotJ committed:
Fixed issue from including burger div when selecting areas for padding in site

SirDotJ committed:
Attempt at fixing tablet work section sizing issues on low width's in landscape mode, fix of experience section on mobile

SirDotJ committed:
Burger button is completed and animated.

SirDotJ committed:
Reminder is implemented. Appears every two minutes. Can be changed in reminder.js

SirDotJ committed:
MAKE IT STOP MAKE IT STOP

SirDotJ committed:
Contact coloring of letters using javascript. Also recolor of provided svg using Codepen's CSS filter generator from black to hex.

SirDotJ committed:
Added strict to all js files so far.

SirDotJ committed:
Fixed some issues with experience period text (missing border-radius for display, text alignement for desktop and tablet's period text

SirDotJ committed:
Fixed some issues with experience period text (missing border-radius for display, text alignement for desktop and tablet's period text

SirDotJ committed:
Refactor as preparation for filling in experience using javascript (to avoid repetition).

SirDotJ committed:
Merge branch 'master' into task-3

# Conflicts:
#	css/contact.css
#	css/experience.css
#	css/site.css

SirDotJ committed:
Working experience switching. Ready for animation.

SirDotJ committed:
Got rid of max height for experience so that it shows correctly on mobile

SirDotJ committed:
Got rid of max height for experience so that it shows correctly on mobile

SirDotJ committed:
Re enabled reminder for review.

SirDotJ committed:
Merge pull request #2 from SirDotJ/task-3

Результат выполнения задания №3

WelcomeDev commented on assets/images/list_point_orange.svg:
такое лучше делать через css)

WelcomeDev commented on css/header.css:
```diff
@@ -0,0 +1,100 @@
+    background: var(--orange-400);
+}
+
+#burger-menu {
```
![image](https://github.com/SirDotJ/WS-Frontend-Capibara/assets/65521226/1f1cfe2f-3637-480c-bf1d-417fcbbb6875)

А где блюр слева, по дизайну должен быть

WelcomeDev commented on js/reminder.js:
```diff
@@ -0,0 +1,17 @@
+const enable = true;
+const timerIntervalMS = 120000;
+
+const reminder = document.getElementById("reminder")
```
прикольно выкатывается, давай только первый интервал выкатывания сократим до пары секунд

WelcomeDev commented on js/experienceDisplaySwitch.js:
```diff
@@ -0,0 +1,84 @@
+    "Frontend Trainee"
+]
+
+const tasks = [
```
предлагаю сделать вот так, все одним объектом


```js
const experience = [
   { period: '', role: '',  description: [] }
]
```

WelcomeDev replied:
дергается пэпэц

https://github.com/SirDotJ/WS-Frontend-Capibara/assets/65521226/99aee7dd-c636-4a9e-ac25-93f499d424a9

SirDotJ replied:
Вот с этим переходом не уверен был да, ну попытаюсь переходы тоже добавить

WelcomeDev commented on js/copyOnClick.js:
```diff
@@ -0,0 +1,30 @@
+}
+
+/* Copy phone number */
+const phone = document.getElementById("phone");
```
вижу дублирующийся код 🌚

WelcomeDev replied:
за подсказку к копированию - лайк

WelcomeDev commented on css/hero.css:
```diff
@@ -0,0 +1,187 @@
+.hero {
```
![image](https://github.com/SirDotJ/WS-Frontend-Capibara/assets/65521226/1a4686f7-7056-4131-a191-28d3ffa3b6f5)

где-то плачит обладатель 4к монитора, когда видит это

SirDotJ replied:
что имеется ввиду? просто на фигме обозначен дизайн до 1440px, может тогда ограничить размер для каждого элемента? 

но что тогда делать с волной? она же резко оборвется на краях.

WelcomeDev replied:
По волне я говорил на занятии, можно добавить дополнительную обертку и позиционировать относительно нее тогда ничего не оборвется)

Нужно ограничить контейнер с капибарой до 1440

SirDotJ committed:
SVG for ::before element of experience list is now defined inline instead of infile.

SirDotJ committed:
Shortened code for copyOnClick.js at the expense of passed values in index.html

SirDotJ committed:
Shortened time for reminder to appear on page load

SirDotJ committed:
Changed wave display back to limited to 1440px. However now wave gets cut off at section borders.

SirDotJ committed:
Fixed divider mistake for header elements

SirDotJ committed:
Added blur effect to the background when pulling up burger menu (does not apply to helpful reminder)

SirDotJ committed:
Experience data is now stored in a list of dictionaries, inner html building refactor, defined and added error Message for failed JS filling

SirDotJ committed:
Experience display now gets moved under opened period on mobile

SirDotJ committed:
Added change to layout for experience on mobile with dynamic template-area change when resizing and saved selected option.

SirDotJ committed:
Implemented selected effect for selected experience fields

SirDotJ committed:
Working slide right and slide up fade away for experience display card. No resize for other elements though

SirDotJ committed:
Working fade away for periods on change. But experience section itself still doesn't change height properly.

SirDotJ commented:
Испробовал всё что смог найти и придумать для Experience. Но до сих пор при изменении innerHTML размер внешних элементов не используют transition при изменении своих размеров. Остановился на эффекте исчезновения для периодов и скольжения для отображения опыта.

WelcomeDev commented on js/reminder.js:
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+const enable = true;
+const timerIntervalMS = 120000;
```
константа не говорит о продолжительности, я не знаю сходу, сколько это минут) Минорный коммент

WelcomeDev commented on js/experienceDisplaySwitch.js:
```diff
@@ -0,0 +1,84 @@
+    const taskCount = tasksForPeriod.length;
+    for (let i = 0; i < taskCount; i++) {
+        html +=
+            "<li class=\"test body-2 body-2--normal text--gray-600\">" +
```
используй интерполяцию строк 

```js
const name = 'Vasian';
`Hello ${name}`
```

WelcomeDev commented on js/experienceDisplaySwitch.js:
```diff
@@ -0,0 +1,84 @@
+    let html = "";
+
+    html +=
+        "<img class=\"experience-display__logo\" src=\"assets/images/ws-logo.svg\" alt=\"Логотип White Soft\">" +
```
чтобы не экранировать `"` в js есть еще целых 2 литерала строк. `'` и `

WelcomeDev commented on js/contactColoring.js:
```diff
@@ -0,0 +1,28 @@
+    const length = text.length;
+    let newText = "";
+    for (let i = 0; i < length; i++) {
+        var letter = text.at(i);
```
бан

SirDotJ replied:
ой, упустил var да извиняюсь.

WelcomeDev commented on js/contactColoring.js:
```diff
@@ -0,0 +1,28 @@
+    for (let i = 0; i < length; i++) {
+        var letter = text.at(i);
+        if (i % 2 === 0) { // color_1
+            letter = `<span style="color: ${color_1}">` + letter + `</span>`
```
если ты знаешь про интерполяцию, то зачем писать через +....

SirDotJ replied:
остатки с первых попыток написания, узнал об интерполяции в середине разработки

WelcomeDev commented on js/copyOnClick.js:
```diff
@@ -0,0 +1,30 @@
+/* Utility Functions */
+const copyField = async(tag) => {
+    const content = tag.innerText.replaceAll(" ", " ");
+    await navigator.clipboard.writeText(content);
```
а зачем тут await? Если микротаску не ожидать, то она все равно выполнится. Ожидание нужно, когда нужно получить значение или как-то оповестить пользователя 🤔 Но я не вижу тут никаких смен состояния

SirDotJ replied:
исправил но изначально добавил по причине: IDE жаловался, не знал что await означал на то время, благодарю за пояснение.

WelcomeDev commented on js/copyOnClick.js:
```diff
@@ -0,0 +1,30 @@
+const email = document.getElementById("email");
+const emailCopyTooltip = document.getElementById("email-copy-tooltip")
+const copyEmail = async () => {
+    await copyField(email);
```
тут чтобы не писать await, можно сделать copyField(email).then(()=>playFade...)

sergio2O1 commented:
1. для меню добавь тени слева, а то сливается как то с контентом на заднем плане, в дизайне она есть
2. в момент перехода по каждому из пунктов меню, думаю что нужно еще менюшку закрывать

![image](https://github.com/SirDotJ/WS-Frontend-Capibara/assets/105034483/c2663177-ce86-4860-b37c-32fb20ed8b01)

sergio2O1 commented:
иконки наверх уехали, плюс еще между иконкой и текстом с почтой/телефоном нет отступа

![image](https://github.com/SirDotJ/WS-Frontend-Capibara/assets/105034483/04371dd8-e91b-4c84-89e3-fcb40f8df366)

sergio2O1 commented on css/contact.css:
```diff
@@ -0,0 +1,117 @@
+    filter: invert(71%) sepia(60%) saturate(855%) hue-rotate(2deg) brightness(108%) contrast(103%);
+}
+
+#email-copy-tooltip, #phone-copy-tooltip {
```
На будущее: как правило, стили нужно указывать по селектору класса, id больше нужен для js, ну или как в данном случае, чтобы перейти к нужной секции при клике на ссылку

хотя здесь по хорошему нужно класс использовать, и просвоить его каждому из этих элементов

sergio2O1 commented on css/contact.css:
```diff
@@ -0,0 +1,117 @@
+
+/* Desktop */
+@media (min-width: 1024px) {
+    #email-copy-tooltip {
```
здесь тоже можно и нужно через класс делать, свойства для email и телефона можно разруливать через модификаторы по БЭМу

sergio2O1 commented on js/contactColoring.js:
```diff
@@ -0,0 +1,28 @@
+    let newText = "";
+    for (let i = 0; i < length; i++) {
+        var letter = text.at(i);
+        if (i % 2 === 0) { // color_1
```
подобные конструкции

```js
if (condition) {
    res = 'TRUE_RESULT'
} else {
    res = 'FALSE_RESULT'
}
```

можно записать через тернарку

```js
res = condition
    ? 'TRUE_RESULT'
    : 'FALSE_RESULT'
```

SirDotJ replied:
О тернарной операции знаю но не думал что он в JavaScript реализован. Круто.

SirDotJ committed:
Fixed wave: now it extends without having to make hero section big.

SirDotJ committed:
Added comments to reminder timings.

SirDotJ committed:
Fixed half attempt at interpolation and accidental var inclusion in contactColoring.
(experienceDisplaySwitch already uses interpolation, don't remember when that happened)

SirDotJ committed:
copyOnClick fixes for incorrect await usage.

SirDotJ committed:
Burger menu: added missed drop-shadow 2x1, menu now closes on navigation click

SirDotJ committed:
Ternary operation used instead of simple if-else in contactColoring.

SirDotJ committed:
Small refactor on contact section to use id for js and class for css. Displayed svgs for contacts are now centered to text and do not go out of bounds.

SirDotJ committed:
Fixed issue that came up due to span coloring: phone now gets copied without new lines.
